### PR TITLE
Automatic port of System.Security docs

### PR DESCRIPTION
**Pull request by carlossanlop**

### CODE REVIEW

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2Collection.xml [hunk 1]
```diff
@@ -1199,9 +1214,29 @@
...
+ PEM-encoded items with a CERTIFICATE PEM label will be imported. PEM items with other labels will be ignored.
+            
+
+ More advanced scenarios for loading certificates and can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
+            
+
+
+
+          ]]></format>
+        </remarks>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2Collection.xml:1236]:
```suggestion
        <remarks>
          <format type="text/markdown"><![CDATA[

## Remarks

 See <xref:System.IO.File.ReadAllText(System.String)> for additional documentation about exceptions that can be thrown.

 PEM-encoded items with a CERTIFICATE PEM label will be imported. PEM items with other labels will be ignored.

 More advanced scenarios for loading certificates and can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.

          ]]></format>
        </remarks>
```

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2Collection.xml [hunk 2]
```diff
@@ -1172,9 +1172,24 @@
...
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2Collection.xml:1191]:
```suggestion
        <remarks>
          <format type="text/markdown"><![CDATA[

## Remarks

 PEM-encoded items with a CERTIFICATE PEM label will be imported. PEM items with other labels will be ignored.
            
 More advanced scenarios for loading certificates and can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.

          ]]></format>
        </remarks>
```

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml [hunk 3]
```diff
@@ -1117,11 +1201,43 @@
...
+
+ Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with an acceptable label is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
+            
+
+ For password protected PEM-encoded keys, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromEncryptedPemFile(System.String,System.ReadOnlySpan{System.Char},System.String)> to specify a password.
+            
+
+
+
+          ]]></format>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml:1233]:
```suggestion
          <format type="text/markdown"><![CDATA[

## Remarks

 See <xref:System.IO.File.ReadAllText(System.String)> for additional documentation about exceptions that can be thrown.

 The SubjectPublicKeyInfo from the certificate determines what PEM labels are accepted for the private key.
              For RSA certificates, accepted private key PEM labels are "RSA PRIVATE KEY" and "PRIVATE KEY".
              For ECDSA certificates, accepted private key PEM labels are "EC PRIVATE KEY" and "PRIVATE KEY".
              For DSA certificates, the accepted private key PEM label is "PRIVATE KEY".

PEM-encoded items that have a different label are ignored.

 Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with an acceptable label is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.

 For password protected PEM-encoded keys, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromEncryptedPemFile(System.String,System.ReadOnlySpan{System.Char},System.String)> to specify a password.

          ]]></format>
```

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml [hunk 4]
```diff
@@ -1088,11 +1145,38 @@
...
+ If the PEM-encoded certificate and private key are in the same text, use the same string for both `certPem` and `keyPem`, such as:
+              <code> CreateFromPem(combinedCertAndKey, combinedCertAndKey); </code> Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with an acceptable label is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
+            
+
+ For password protected PEM-encoded keys, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromEncryptedPem(System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char})> to specify a password.
+            
+
+
+
+          ]]></format>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml:1174]:
```suggestion
          <format type="text/markdown"><![CDATA[

## Remarks

 The SubjectPublicKeyInfo from the certificate determines what PEM labels are accepted for the private key.
              For RSA certificates, accepted private key PEM labels are "RSA PRIVATE KEY" and "PRIVATE KEY".
              For ECDSA certificates, accepted private key PEM labels are "EC PRIVATE KEY" and "PRIVATE KEY".
              For DSA certificates, the accepted private key PEM label is "PRIVATE KEY".
            
PEM-encoded items that have a different label are ignored.

 If the PEM-encoded certificate and private key are in the same text, use the same string for both `certPem` and `keyPem`, such as:
              <code> CreateFromPem(combinedCertAndKey, combinedCertAndKey); </code> Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with an acceptable label is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.

 For password protected PEM-encoded keys, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromEncryptedPem(System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char})> to specify a password.

          ]]></format>
```

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml [hunk 5]
```diff
@@ -1057,12 +1084,42 @@
...
+
+ Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with the label "ENCRYPTED PRIVATE KEY" is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
+            
+
+ For PEM-encoded keys without a password, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromPemFile(System.String,System.String)>.
+            
+
+
+
+          ]]></format>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml:1114]:
```suggestion
          <format type="text/markdown"><![CDATA[

## Remarks

 See <xref:System.IO.File.ReadAllText(System.String)> for additional documentation about exceptions that can be thrown.
            
 Password protected PEM-encoded keys are always expected to have the PEM label "ENCRYPTED PRIVATE KEY".
            
PEM-encoded items that have a different label are ignored.

 Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with the label "ENCRYPTED PRIVATE KEY" is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
            
 For PEM-encoded keys without a password, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromPemFile(System.String,System.String)>.
            
          ]]></format>
```

File: xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml [hunk 6]
```diff
@@ -1026,12 +1028,37 @@
...
+ If the PEM-encoded certificate and private key are in the same text, use the same string for both `certPem` and `keyPem`, such as:
+              <code> CreateFromEncryptedPem(combinedCertAndKey, combinedCertAndKey, theKeyPassword); </code> Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with the label "ENCRYPTED PRIVATE KEY" is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
+            
+
+ For PEM-encoded keys without a password, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromPem(System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char})>.
+            
+
+
+
+          ]]></format>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography.X509Certificates/X509Certificate2.xml:1055]:
```suggestion
          <format type="text/markdown"><![CDATA[

## Remarks

 Password protected PEM-encoded keys are always expected to have the PEM label "ENCRYPTED PRIVATE KEY".
            
PEM-encoded items that have a different label are ignored.

 If the PEM-encoded certificate and private key are in the same text, use the same string for both `certPem` and `keyPem`, such as:
              <code> CreateFromEncryptedPem(combinedCertAndKey, combinedCertAndKey, theKeyPassword); </code> Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the the first certificate with a CERTIFICATE label is loaded. For the private key, the first private key with the label "ENCRYPTED PRIVATE KEY" is loaded. More advanced scenarios for loading certificates and private keys can leverage <xref:System.Security.Cryptography.PemEncoding> to enumerate PEM-encoded values and apply any custom loading behavior.
            
 For PEM-encoded keys without a password, use <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.CreateFromPem(System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char})>.
            
          ]]></format>
```

**Review by pgovind — APPROVED** [review]
This is great documentation, very thorough!

**Comment by carlossanlop** [comment]
> This is great documentation, very thorough!

Agreed! cc @bartonjs

**Pull request merged**